.github/workflows/ci-checks.yml

```diff
@@ -0,0 +1,81 @@
+name: CI Checks
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+env:
+  IMAGE_NAME: "fruits-app"
+  BASE_IMAGE: "registry.access.redhat.com/ubi8/openjdk-11"
+
+
+jobs:
+  build-maven:
+    runs-on: ubuntu-latest
+    steps:
+       # Checkout Knative service deploy action github repository
+      - name: Checkout Knative service deploy action
+        uses: actions/checkout@v2
+        with:
+          path: "kn-service-deploy"
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Checkout example code
+        id: git_checkout
+        uses: actions/checkout@v2
+        with:
+          repository: kameshsampath/kn-service-action-example
+          path: 'fruits-app'
+
+      - name: Maven Build
+        id: build_app
+        working-directory: fruits-app
+        run: mvn -B -DskipTests clean package --file pom.xml
+
+      - name: Cache Maven packages
```

Comment:
If you're going to upload this cache, at some point you have to download it too (before the maven build) or there's no point